### PR TITLE
Update dependencies, handle redirects, prepare for publishing

### PR DESCRIPTION
**Pull request by jbearer**

### CODE REVIEW

File: Cargo.toml
Comment (nyospe) [inline Cargo.toml:1]: Unfortunate that we're still sick with this vulnerable version of async-tungstenite from tide...

File: src/socket.rs [hunk 1]
```diff
@@ -15,53 +18,76 @@ use futures::{
...
+            };
+            if let WsError::Http(res) = &err {
+                if (301..=308).contains(&u16::from(res.status())) {
+                    if let Some(location) = res
+                        .headers()
+                        .get("location")
+                        .and_then(|header| header.to_str().ok())
+                    {
+                        tracing::info!(from = %self.url, to = %location, "WS handshake following redirect");
+                        self.url.set_path(location);
```
Comment (nyospe) [inline src/socket.rs:84]: Could we maybe change the default to use binary?

**Review by nyospe — APPROVED** [review]
I think everything looks good. Would be nice to change the streaming default

File: Cargo.toml
Comment (jbearer) [reply Cargo.toml:1]: Indeed. A problem for another day...

File: src/socket.rs (hunk 1 above)
Comment (jbearer) [reply src/socket.rs:84]: Ah, this should actually respect the preference set in the builder. I'll fix it

**Pull request merged**